feat:  add C implementation for `stats/base/dists/binomial/kurtosis`

0PrashantYadav0 opened the pull request:
Resolves #3467

## Description

> What is the purpose of this pull request?

This pull request:

-   adds C implementation for @stdlib/stats/base/dists/binomial/kurtosis along with relevant tests, docs, examples and benchmarks.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #3467

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

Neerajpathak07 committed:
refactor: update `stats/base/dnanmeanpn` native addon from C++ to C

PR-URL: https://github.com/stdlib-js/stdlib/pull/4110
Reviewed-by: Athan Reines <kgryte@gmail.com>

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: passed
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 committed:
Merge branch 'develop' of https://github.com/0PrashantYadav0/stdlib into develop

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 committed:
Merge branch 'develop' of https://github.com/0PrashantYadav0/stdlib into develop

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 committed:
Merge branch 'stdlib-js:develop' into develop

0PrashantYadav0 committed:
feat: add C implementation for binomail kurtosis

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter reviewed (changes requested):
Instead of `int`, can we please type `n` as `int32_t`? The latter is part of the C standard library's `<stdint.h>` header and ensures portability across different systems by defining exact-width integer types. See e.g. https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/fibonacci for a package with a `int32_t` parameter.

Notice that this PR also contains instances such as in `kurtosis.h` where `n` is still `double`-typed, too, so these will have to be updated as well.

Would you be able to please update all your `binomial/*` PRs accordingly?

0PrashantYadav0 commented:
@Planeshifter i am having trouble in doing so as i have to use `STDLIB_MATH_BASE_NAPI_MODULE_ID_D` which is not implemented in `stdlib/math/base/napi/binary.h` . Do i have to implement this as my own, if so then how can i do this ?

0PrashantYadav0 commented:
@Planeshifter By changing the double into int32_t i have remove some test cases as it will provide undefined answer.

0PrashantYadav0 committed:
chore: replace double to int32_t for n

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 committed:
chore: replace double to int32_t for n

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
---

0PrashantYadav0 commented:
@Planeshifter I have made the required changes in this pr please review and if its ok i will make similar changes in all my prs.

0PrashantYadav0 committed:
fix: edited examples

Signed-off-by: JoyBoy <144602492+0PrashantYadav0@users.noreply.github.com>

0PrashantYadav0 commented:
/stdlib update-copyright-years

stdlib-bot committed:
chore: update copyright years

kgryte commented on lib/node_modules/@stdlib/math/base/napi/binary/README.md:
```diff
@@ -820,6 +820,29 @@ The macro expects the following arguments:
 
 When used, this macro should be used **instead of** `NAPI_MODULE`. The macro includes `NAPI_MODULE`, thus ensuring Node-API module registration.
 
+#### STDLIB_MATH_BASE_NAPI_MODULE_ID_D( fcn )
```
@0PrashantYadav0 Would you mind submitting the changes to `math/base/napi/binary` as a separate PR? This will help us ensure clean commits and an accurate changelog.

kgryte replied:
We can first merge that PR before merging this one after merging in the latest changes on `develop`.

0PrashantYadav0 replied:
sure i will open new pr for this

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/README.md:
```diff
@@ -141,6 +141,103 @@ for ( i = 0; i < 10; i++ ) {
+
+#### stdlib_base_dists_binomial_kurtosis( n, p )
+
+Returns the kurtosis of a binomial distribution.
```
```suggestion
Returns the [excess kurtosis][kurtosis] of a [binomial][binomial-distribution] distribution with number of trials `n` and success probability `p`.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/README.md:
```diff
@@ -141,6 +141,103 @@ for ( i = 0; i < 10; i++ ) {
+
+The function accepts the following arguments:
+
+-   **n**: `[in] int` number of trials.
```
```suggestion
-   **n**: `[in] int32_t` number of trials.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/README.md:
````diff
@@ -141,6 +141,103 @@ for ( i = 0; i < 10; i++ ) {
+-   **p**: `[in] double` success probability.
+
+```c
+double stdlib_base_dists_binomial_kurtosis( const int n, const double p );
````
```suggestion
double stdlib_base_dists_binomial_kurtosis( const int32_t n, const double p );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/benchmark/c/benchmark.c:
```diff
@@ -0,0 +1,143 @@
+	int i;
+
+	for ( i = 0; i < 100; i++ ) {
+		n[ i ] = stdlib_base_ceil( random_uniform( 0, 100 ) );
+		p[ i ] = random_uniform( 0, 1 );
```
```suggestion
		n[ i ] = stdlib_base_ceil( random_uniform( 0.0, 100.0 ) );
		p[ i ] = random_uniform( 0.0, 1.0 );
```

`random_uniform` expects doubles, so we add the decimal points.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/README.md:
```diff
@@ -141,6 +141,103 @@ for ( i = 0; i < 10; i++ ) {
+    int i;
+
+    for ( i = 0; i < 25; i++ ) {
+        n = stdlib_base_ceil( random_uniform( 0, 100 ) );
+        p = random_uniform( 0, 1 );
```
```suggestion
        n = stdlib_base_ceil( random_uniform( 0.0, 100.0 ) );
        p = random_uniform( 0.0, 1.0 );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/examples/c/example.c:
```diff
@@ -0,0 +1,44 @@
+	int i;
+
+	for ( i = 0; i < 25; i++ ) {
+		n = stdlib_base_ceil( random_uniform( 0, 100 ) );
+		p = random_uniform( 0, 1 );
```
```suggestion
		n = stdlib_base_ceil( random_uniform( 0.0, 100.0 ) );
		p = random_uniform( 0.0, 1.0 );
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/include/stdlib/stats/base/dists/binomial/kurtosis.h:
```diff
@@ -0,0 +1,40 @@
+#endif
+
+/**
+* Returns the kurtosis of a binomial distribution.
```
```suggestion
* Returns the excess kurtosis of a binomial distribution with number of trials `n` and success probability `p`.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/lib/native.js:
```diff
@@ -0,0 +1,59 @@
+// MAIN //
+
+/**
+* Returns the kurtosis of a binomial distribution.
```
```suggestion
* Returns the excess kurtosis of a binomial distribution with number of trials `n` and success probability `p`.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,49 @@
+#include <stdint.h>
+
+/**
+* Returns the kurtosis of a binomial distribution.
```
```suggestion
* Returns the excess kurtosis of a binomial distribution with number of trials `n` and success probability `p`.
```

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,49 @@
+	if (
+		stdlib_base_is_nan( n ) ||
+		stdlib_base_is_nan( p ) ||
+		n < 0.0 ||
```
```suggestion
		n < 0 ||
```

`n` is an integer.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,49 @@
+*/
+double stdlib_base_dists_binomial_kurtosis( const int32_t n, const double p ) {
+	if (
+		stdlib_base_is_nan( n ) ||
```
```suggestion
```

`n` is an integer.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,49 @@
+
+#include "stdlib/stats/base/dists/binomial/kurtosis.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/constants/float64/pinf.h"
```
This constant is not used. The `manifest.json` file should also be updated accordingly.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,49 @@
+		stdlib_base_is_nan( n ) ||
+		stdlib_base_is_nan( p ) ||
+		n < 0.0 ||
+		n >= INT32_MAX ||
```
If `n` is `int32_t`, in what circumstance could it ever exceed the max `int32`?

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,49 @@
+		return 0.0 / 0.0;
+	}
+	double pq = p * ( 1.0 - p );
+	return ( 1.0 - (6.0 * pq ) ) / ( n * pq );
```
```suggestion
	return ( 1.0 - ( 6.0 * pq ) ) / ( n * pq );
```

Use consistent spacing.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/manifest.json:
```diff
@@ -0,0 +1,85 @@
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/math/base/assert/is-nonnegative-integer",
+        "@stdlib/constants/float64/pinf",
+        "@stdlib/math/base/special/ceil"
```
This dep is not used in the implementation.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/manifest.json:
```diff
@@ -0,0 +1,85 @@
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/math/base/assert/is-nonnegative-integer",
+        "@stdlib/constants/float64/pinf",
```
This dep is not used.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,49 @@
+#include "stdlib/stats/base/dists/binomial/kurtosis.h"
+#include "stdlib/math/base/assert/is_nan.h"
+#include "stdlib/constants/float64/pinf.h"
+#include "stdlib/math/base/assert/is_nonnegative_integer.h"
```
This dep is not used.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/manifest.json:
```diff
@@ -0,0 +1,85 @@
+      "dependencies": [
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/math/base/assert/is-nonnegative-integer",
```
This dep is not used.

kgryte commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/test/test.native.js:
```diff
@@ -0,0 +1,118 @@
+		}
+		else {
+			delta = abs( y - expected[i] );
+			tol = 100.0 * EPS * abs( expected[i] );
```
Why is this tolerance so high? The implementation is simple arithmetic and does not involve transcendentals.

0PrashantYadav0 replied:
This is due point precision error. It shows error upto 2 decimal points thats why it has high tolerance.

0PrashantYadav0 committed:
chore: changes request in code reviews

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: passed
---

0PrashantYadav0 committed:
Merge branch 'binomial/kurtosis' of https://github.com/0PrashantYadav0/stdlib into binomial/kurtosis

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: passed
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@kgryte I open pr for binary as well as for ternary implementation, #4522 and #4523 . Please review and merge them asap so that i can move forward and implement these in all the prs of binomial/* .

0PrashantYadav0 committed:
chore: added spaces

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 committed:
fix: added function

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: passed
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 committed:
chore: catchup with last commit

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: passed
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

0PrashantYadav0 commented:
@Planeshifter @kgryte  Please review this, i have implemented `ID_D` as mention earlier.

Planeshifter reviewed (commented):
@0PrashantYadav0 Would you mind resolving the merge conflicts in this PR. please?

0PrashantYadav0 commented:
/stdlib merge

0PrashantYadav0 committed:
chore: merge to develop

0PrashantYadav0 committed:
fix: changes in benchmarks

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: failed
---

0PrashantYadav0 commented:
@Planeshifter Done please review and merge. Thank you.

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/src/main.c:
```diff
@@ -0,0 +1,45 @@
+#include <stdint.h>
+
+/**
+* Returns the kurtosis of a binomial distribution.
```
```suggestion
* Returns the excess kurtosis of a binomial distribution with number of trials `n` and success probability `p`.
```

Planeshifter commented on lib/node_modules/@stdlib/stats/base/dists/binomial/kurtosis/manifest.json:
```diff
@@ -0,0 +1,79 @@
+      "libpath": [],
+      "dependencies": [
+        "@stdlib/math/base/napi/binary",
+        "@stdlib/math/base/assert/is-nan",
+        "@stdlib/math/base/special/ceil"
```
```suggestion
        "@stdlib/math/base/assert/is-nan"
```

Planeshifter committed:
chore: apply suggestions from code review

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Great, thanks for resolving the merge conflicts! 

Did a review pass and PR looks to be in good shape overall now. Had to apply to edits that @kgryte initially flagged but which were not resolved; please ensure to only resolve comments once addressed.

I will go ahead and merge your PR shortly. Thanks again for all your contributions!
